### DISCUSSION TITLE
What parts (and how much) of the Personal Use Area does Nerd Fonts use?

### DISCUSSION THREAD

**Discussion by eamonburns** [ryanoasis/nerd-fonts#1539] [Q&A] [answered]
I am wanting to create my own font that utilizes some of the PUA space as a hobby project, and I would like it not to conflict with Nerd Fonts if possible. Are there any sections of the PUA that Nerd Fonts does not use?

**Comment by ChiefMikeK** [comment]
It's best to use a block in [Plane 15](https://en.wikipedia.org/wiki/Plane_\(Unicode\)#Private_Use_Area_Planes ) that others are not using see wikipedia and also [Under-ConScript Unicode Registry](https://www.kreativekorp.com/ucsur/ )

**Comment by Finii** [comment, marked as answer]
For me it is unclear if you mentioning `PUA` means 
* **the** PUA, i.e. `E000` - `F8FF`
* any PUA, including `PUA-A` and `PUA-B` (`F'0000` - `F'FFFD`; `10'000` - `10'FFFD`)

I believe CSUR mentioned by @ChiefMikeK has no relevance here, as it also 'reserves' codepoints used by Nerd Fonts :-D

> The [ConScript Unicode Registry (CSUR)](http://www.evertype.com/standards/csur/) is a project led by John Cowan and Michael Everson to coordinate the assignment of blocks out of the Unicode Private Use Area (E000-F8FF and F0000-10FFFF) to constructed/artificial scripts, including scripts for constructed/artificial languages.

An overview of the Nerd Font codepoints is here:
https://github.com/ryanoasis/nerd-fonts/wiki/Glyph-Sets-and-Code-Points#overview

Keep in mind that some people might have problems with 'more than four hex-digit' codepoints, i.e. can not use them, but these cases are getting more rare in the last time, so that we put the Material Design icons in the 5 digit range (`PUA-A`) with Nerd Font v3.

**Reply by eamonburns** [reply]
By `PUA`, I think I mean _any `PUA`_. But I am not very well versed in the Unicode standard. Thank you for linking that Wiki page. Based on the Wiki page, it looks like most of `PUA-A`, and all of `PUA-B` are open.

**Reply by Finii** [reply]
Correct.
There are some gaps in the 4 digit range still (i.e. the `PUA` PUA).

In `PUA-A` it might be good to put some space between the end of MaterialDesign and your codepoints, as MD might grow in the future. We put it there specifically to allow 'unbounded growth' ;-)

**Reply by eamonburns** [reply]
Yeah, I am thinking of starting it at `FF000`
